File: back-end/back-end/src/main/java/com/datn/backend/service/HinhAnhSanPhamService.java
package com.datn.backend.service;

import com.datn.backend.model.san_pham.HinhAnh;

import java.util.List;

public interface HinhAnhSanPhamService {

    List<HinhAnh> getByMauSac(String tenMau, int sanPhamID);

    List<HinhAnh> getImgsOf1ProductColor(int prodId, int colorId);

    List<String> getAllUrlBySanPhamAndMauSac(int productId, int colorId);
}

File: admin-fe/src/app/service/hinh-anh-san-pham.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { Observable } from "rxjs";

import { HinhAnh } from "../model/class/hinh-anh.class";

@Injectable({
  providedIn: "root",
})
export class HinhAnhSanPhamService {
  private readonly apiURL = "http://localhost:8080/hinh-anh-sp";

  constructor(private http: HttpClient) {}

  public getAll(tenMau: string, sanPhamID: number): Observable<HinhAnh[]> {
    let params = `?tenMau=${tenMau}&sanPhamID=${sanPhamID}`;
    return this.http.get<HinhAnh[]>(`${this.apiURL}/get-by-ms${params}`);
  }

  public getImgsOf1ProductColor(
    prodId: number,
    colorId: number
  ): Observable<HinhAnh[]> {
    return this.http.get<HinhAnh[]>(
      `${this.apiURL}/by-product-color/${prodId}/${colorId}`
    );
  }
}
